### PR TITLE
updatecheck: Try to revive on dotcom

### PR DESCRIPTION
**Pull request by eseliger**
In an attempt to revive pings, this PR does:

- Bump timeout to 30m from 10m
- Don’t fail insertion of the pings data event for timeouts
- Separate timeout for collecting the data and sending the data
- Add some better error messages by wrapping more errors in useful pointers.

Later, we should:
- Investigate improving the queries
- Investigate increase in pings data

## Test plan

CI / code reviews.

### CODE REVIEW

File: cmd/frontend/internal/app/updatecheck/client.go [hunk 1]
```diff
@@ -729,15 +729,19 @@ func updateBody(ctx context.Context, logger log.Logger, db database.DB) (io.Read
...
-		URL:             "",
-		AnonymousUserID: "backend",
-		Source:          "BACKEND",
-		Argument:        contents,
-		Timestamp:       time.Now().UTC(),
-	})
+	err = db.EventLogs().Insert(
+		// We always want to insert this event, even when the context is expired when
+		// we get here.
+		context.Background(),
```
Comment (camdencheek) [inline cmd/frontend/internal/app/updatecheck/client.go:735]: Do we want to insert incomplete events? We currently do not fail if the context expires and some parts of the ping fail to be collected, we just log a message.

Comment (eseliger) [reply cmd/frontend/internal/app/updatecheck/client.go:735]: if there's another error that would not fail this insertion, too, we would also have incomplete data so I assumed that was the desired behavior 🤔 Maybe not though?

Comment (camdencheek) [reply cmd/frontend/internal/app/updatecheck/client.go:735]: Maybe?

Comment (dadlerj) [reply cmd/frontend/internal/app/updatecheck/client.go:735]: Yeah I actually agree that we should hard fail on incomplete data for sourcegraph.com... For self-hosted, on the other hand, it'd be great to get a partial dataset!

Just to check—would it be possible to know if some set of the queries timed out? E.g. would normal Go error handling... catch that, somehow? I'm wondering if we could add a new data field to the ping payload called "ping_errors" or something that aggregates all timeouts and errors.

**Pull request closed without merging**